File: web/firebase-config.js
// Firebase Configuration for dndn-86c02
const firebaseConfig = {
  apiKey: "AIzaSyARZMki7TeMsoHSDiJUTdt2bUGmxVYEtP4",
  authDomain: "dndn-86c02.firebaseapp.com",
  projectId: "dndn-86c02",
  storageBucket: "dndn-86c02.firebasestorage.app",
  messagingSenderId: "532869168095",
  appId: "1:532869168095:web:3af71d7ff6e1d81aba226a",
  measurementId: "G-YGMT61SG32"
};

// Initialize Firebase
import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getFirestore } from 'firebase/firestore';
import { getStorage } from 'firebase/storage';
import { getMessaging } from 'firebase/messaging';

const app = initializeApp(firebaseConfig);

export const auth = getAuth(app);
export const db = getFirestore(app);
export const storage = getStorage(app);
export const messaging = getMessaging(app);
export default app;

File: lib/config/firebase_config.dart
import 'package:firebase_core/firebase_core.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import '../firebase_options.dart';

class FirebaseConfig {
  static FirebaseApp? _app;
  static FirebaseAuth? _auth;
  static FirebaseFirestore? _firestore;

  // تهيئة Firebase باستخدام الإعدادات الجديدة
  static Future<void> initialize() async {
    if (_app == null) {
      _app = await Firebase.initializeApp(
        options: DefaultFirebaseOptions.currentPlatform,
      );
    }
  }

  // الحصول على مثيل المصادقة
  static FirebaseAuth get auth {
    _auth ??= FirebaseAuth.instance;
    return _auth!;
  }

  // الحصول على مثيل قاعدة البيانات
  static FirebaseFirestore get firestore {
    _firestore ??= FirebaseFirestore.instance;
    return _firestore!;
  }

  // التحقق من حالة الاتصال
  static bool get isInitialized => _app != null;
}
